manifest.json

```diff
@@ -1,14 +1,16 @@
 {
-  "manifest_version": 2,
+  "manifest_version": 3,
   "name": "__MSG_extensionName__",
   "description": "__MSG_extensionDescription__",
   "version": "1.38.0",
   "version_name": "1.38.0-prerelease",
   "default_locale": "en",
   "background": {
+    "service_worker": "background/background.js",
```

Comment:
A lot happens every time the service worker is reloaded, which for us is every minute[^1]. It would be a good idea to minimize any unnecessary computations, especially getting all the addon manifests as I said before.

[^1]: See [code search (2 files)](https://github.com/search?q=repo%3AScratchAddons%2FScratchAddons+periodInMinutes&type=code)

Reply:
The service worker will not reload 60 times per minute, in any reasonable desktop browser

Reply:
For example, AFAIK Chrome will keep the service worker active for 5 minutes as long as there's enough memory

Reply:
> For example, AFAIK Chrome will keep the service worker active for 5 minutes as long as there's enough memory

You sure? I wrote some debugging code to count each time the addon manifests were loaded from disk into memory, turned Scratch Addons on, and read some articles while I waited and after about 30 minutes, the service worker had reloaded 30 times. Unless maybe I just need to do more extensive testing.

By the way, I just had to insert this at line 37 in `load-addon-manifests.js` to count the reloads.
```js
chrome.storage.local.set({ restarts: ((await chrome.storage.local.get("restarts")).restarts || 0) + 1 });
```

The timer, however long it is, probably resets every time you visit a page on Scratch, so if you want to test the service worker's lifecycle in the case where Scratch Addons isn't doing much of anything, I'd advise you browse on other websites for a little bit while measuring the extension's activity.

Even if it stays active for longer sometimes, Google says it's best not to keep a service worker alive indefinitely.

Anyway, browser extensions are just the kind of thing that we generally expect to use very little system resources, so we can probably do better than rebuilding data that we already have only to never use it most of the time (whenever the user is not on Scratch or stays on the same page for an extended period of time, like when playing a game on Scratch). [One of my issues](https://github.com/ScratchAddons/ScratchAddons/issues/6355#issuecomment-1619429386) got closed for this reason.

All of that might turn out to just be my opinion, but I recommend we try.

<sub>Windows 11, Edge 124, 16 GB RAM, commit https://github.com/ScratchAddons/ScratchAddons/commit/67d8edfacac18eeb4b3554e990aa421d89e617c2</sub>

Reply:
I would say we look into this in the future

Reply:
Chrome knows if we have a periodic alarm that runs every 1 minute. If it decides that it's better for the user (and system resources) to kill and restart the service worker 60 times per minute, I find that controversial.

Reply:
I'm looking into using `chrome.storage.session` to save addon manifests to memory in Chrome 102+. Extension session storage is cleared when the browser is restarted, or the extension is updated, reloaded, or disabled and reenabled. If I find a solution, I'll open a PR when I'm done.

Reply:
Looks like we'll also need to do this with the localized strings.

Reply:
Also, the cache should be disabled in development mode, if the extension updates to a new version, and if the value of the "show addon names in English" setting changed.

Reply:
> Also, the cache should be disabled in development mode, if the extension updates to a new version, and if the value of the "show addon names in English" setting changed.

Extension session storage is cleared when the browser is restarted, or the extension is updated, reload, or disabled and then reenabled, according to my testing. This might not apply to all browsers or environments, though. Depending on the behavior, we may never have to clear session storage ourselves at all.

Reply:
But clearing it anyway would still be safe. We could clear on `chrome.runtime.onInstalled`, which fires every browser restart or extension install, update, reload, or enable.